### PR TITLE
Save current step in local storage(fixed)

### PR DESCRIPTION
**Pull request by maryna-velychko**

### CODE REVIEW

File: src/app/(auth)/signup/page.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+  useEffect(() => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_NAME);
+    if (authToken) {
+      router.push('/');
+    }
+    const savedStep = localStorage.getItem('signup_current_step');
+    setCurrentStep(savedStep ? parseInt(savedStep) : 0);
+  }, []);
+
+  useEffect(() => {
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:24]: So here is what we have:
scrolNext() -> state -> localStorage

From Python Zen: flat is better than nexted.
So I would update both current step state and local storage variable from scrollNext function.
This way we have
scrollNext() -> state and localStorage

File: src/app/(auth)/signup/page.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
 export default function SignUp() {
+  const router = useRouter();
   const [currentStep, setCurrentStep] = useState(0);
+
+  useEffect(() => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_NAME);
+    if (authToken) {
+      router.push('/');
+    }
+    const savedStep = localStorage.getItem('signup_current_step');
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:20]: getItem and setItem should use same key. While having them hardcoded in 2 places you introduce a divergent change code smell. You can check shotgun strategy code smell https://refactoring.guru/smells/shotgun-surgery

It is not really critical here, and I still would like you to learn on small examples.

File: src/app/(auth)/signup/page.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ import SignUpEmailConfirmation from "./components/SignUpEmailConfirmation";
 import SignUpChildren from "./components/SignUpChildren";
 import { AUTH_TOKEN_NAME } from "@/app/api.mjs";
 
+const SIGNUP_CURRENT_STEP_KEY = 'signup_current_step';
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:11]: Shoudl be camelCase

File: src/app/(auth)/signup/page.js [hunk 4]
```diff
@@ -17,17 +19,19 @@ export default function SignUp() {
...
-    const savedStep = localStorage.getItem('signup_current_step');
+    const savedStep = localStorage.getItem(SIGNUP_CURRENT_STEP_KEY);
     setCurrentStep(savedStep ? parseInt(savedStep) : 0);
   }, []);
 
-  useEffect(() => {
-    if (currentStep !== 0)
-      localStorage.setItem('signup_current_step', currentStep);
-  }, [currentStep]);
+  const goToNextStep = () => {
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:26]: const goToNextStep = () => {
    localStorage.setItem(SIGNUP_CURRENT_STEP_KEY, nextStep);
    setCurrentStep((prevStep) => prevStep + 1)
  };

File: src/app/(auth)/signup/page.js [hunk 5]
```diff
@@ -1,14 +1,34 @@
...
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { useRouter } from "next/navigation";
 import SignUpAccount from "./components/SignUpAccount";
 import SignUpDetails from "./components/SignUpDetails";
 import SignUpEmailConfirmation from "./components/SignUpEmailConfirmation";
 import SignUpChildren from "./components/SignUpChildren";
+import { AUTH_TOKEN_NAME } from "@/app/api.mjs";
+
+const signupCurrentStepKey = 'signup_current_step';
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:11]: SIGNUP_CURRENT_STEP_KEY, that is syntax for constants in JS.

File: src/app/(auth)/signup/page.js [hunk 6]
```diff
@@ -1,14 +1,34 @@
...
+    if (authToken) {
+      router.push('/');
+    }
+    const signupCurrentStep = localStorage.getItem(signupCurrentStepKey);
+    setCurrentStep(signupCurrentStep ? parseInt(signupCurrentStep) : 0);
+  }, []);
+
+  const goToNextStep = () => {
+    localStorage.setItem(signupCurrentStepKey, currentStep + 1);
+    setCurrentStep((prevStep) => prevStep + 1)
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:28]: So if state variable named currentStep, why here it is named different way?

File: src/app/(auth)/signup/page.js (hunk 5 above)
Comment (maryna-velychko) [reply src/app/(auth)/signup/page.js:11]: your previous comment: "Shoudl be camelCase"

File: src/app/(auth)/signup/page.js [hunk 7]
```diff
@@ -1,14 +1,34 @@
...
+  const router = useRouter();
   const [currentStep, setCurrentStep] = useState(0);
+
+  useEffect(() => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_NAME);
+    if (authToken) {
+      router.push('/');
+    }
+    const signupCurrentStep = localStorage.getItem(SIGNUP_CURRENT_STEP_KEY);
+    setCurrentStep(signupCurrentStep ? parseInt(signupCurrentStep) : 0);
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/page.js:23]: Extract variable for it
signupCurrentStep ? parseInt(signupCurrentStep) : 0

**Pull request merged**